docs/reference/quickstart/full-text-tutorial.asciidoc

```diff
@@ -0,0 +1,428 @@
+[[full-text-tutorial]]
+== Basic searching and filtering in {es}
+++++
+<titleabbrev>Basics: Searching and filtering</titleabbrev>
+++++
+
+This quick start guide is a hands-on introduction to filtering, searching, and aggregrating data using the `_search` API and QueryDSL.
+
+[discrete]
+[[full-text-tutorial-create-index]]
+=== Create an index
+
+We'll use a cooking blog dataset in this tutorial.
+Create the `cooking_blog` index and its mapping to get started.
+
+[source,console]
+----
+PUT /cooking_blog
+{
+  "mappings": {
+    "properties": {
+      "title": {
+        "type": "text",
+        "analyzer": "standard" <1>
+      },
+      "description": {
+        "type": "text"
+      },
+      "author": {
+        "type": "text",
+        "fields": {
+          "keyword": { <2>
+            "type": "keyword",
+            "ignore_above": 256
+          }
+        }
+      },
+      "date": {
+        "type": "date",
+        "format": "yyyy-MM-dd"
+      },
+      "category": {
+        "type": "keyword" 
+      },
+      "tags": {
+        "type": "keyword"
+      },
+      "rating": {
+        "type": "float"
+      }
+    }
+  }
+}
+----
+// TESTSETUP
+<1> The `standard` analyzer is used by default if an `analyzer` isn't specified.
+Note that analyzers are determined per field. Each field can have a different analyzer.
+<2> Uses <<multi-fields,multi-fields>> to index `author` as both `text` and `keyword` types. This will enable both full-text search and strict term matches.
+
+[TIP]
+====
+Full-text search is powered by <<analysis,text analysis>>. 
+Text analysis normalizes and standardizes text data so it can be efficiently stored in an inverted index and searched in near real-time.
```

Comment:
Link to docs defining inverted index here?

Reply:
We don't have this today :sad_panda:.

I would love to have a page about data structures in the [basics section](https://www.elastic.co/guide/en/elasticsearch/reference/current/elasticsearch-intro.html) that summarizes how Elasticsearch stores different data types in optimized structures: inverted index for text, BKD trees for geospatial data, HNSW for vectors etc.

A lot of this core stuff is buried in blogs.

Another disappointing learning doing some of this work is that at some point the Definitive Guide was deprecated but not all of the essential basics were ported over into the docs! 

See https://github.com/elastic/elasticsearch-definitive-guide/blob/master/052_Mapping_Analysis/35_Inverted_index.asciidoc. Need to restore this to the docs ASAP in the right spot.